Project-Euler/Problem016.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Problem 16 - Power digit sum
+ *
+ * @see {@link https://projecteuler.net/problem=16}
+ *
+ * 2¹⁵ = 32768 and the sum of its digits is 3 + 2 + 7 + 6 + 8 = 26.
+ *
+ * What is the sum of the digits of the number 2¹⁰⁰⁰ ?
+ */
+
+/**
+ * Returns the power digit sum of n^pow.
+ *
+ * @param {number} [n=2]
+ * @param {number} [pow=1000]
+ * @returns {number}
+ */
+const powerDigitSum = function (n = 2, pow = 1000) {
+  // The idea is to consider each digit (d*10^exp) separately, right-to-left.
+  // digits = [units, tens, ...]
+
+  const digits = [n]
+  let p = 1
+
+  while (++p <= pow) {
+    let carry = 0
+    for (let exp = 0; exp < digits.length; exp++) {
+      const prod = digits[exp] * n + carry
+      carry = Math.floor(prod / 10)
+      digits[exp] = prod % 10
+    }
+    while (carry > 0) {
+      digits.push(carry % 10)
+      carry = Math.floor(carry / 10)
+    }
+  }
+
+  // (digits are reversed but we only want the sum so it doesn't matter)
+
+  return digits.reduce((prev, current) => prev + current, 0)
+}
+
+console.log('Power digit sum of 2^1000 :', powerDigitSum())
```

Comment:
Remove the console log.

Reply:
I don't understand. If I remove the log then running `node Problem016.js` won't output anything, right ? 

Also, this won't be consistent with what is already comitted (algorithms available on this repo do use console.log). 

I think most people do want to see the output of an algorithm, especially when the input/arguments must fit a specific problem to be solved. But maybe I missed something ?

Reply:
Algorithms are not supposed to log anything. They are supposed to be completely self contained and return all the results, not log them.

Reply:
I agree with that, and that is why by the way the console.log() is outside the exported function that describe and run the "algorithm". 

So, yes of course I can remove it, but in this case :
- Why are there all these console.log() everywhere on that repo already ?
- And why is there a `Please use 'console.log()'` put in evidence in CONTRIBUTING.md ?

Also, I insist that Project Euer problems has specific input and an expected output. It is not just an algorithm like sorting that has nothing to do/log without context/inputs.